app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -75,10 +75,12 @@ class WifiDirectManager @Inject constructor(
   private var shouldRetry = true
 
   // Overall manager of Wifi p2p connections for the module
-  private lateinit var manager: WifiP2pManager
+  private val manager: WifiP2pManager? by lazy(LazyThreadSafetyMode.NONE) {
+    activity.getSystemService(Context.WIFI_P2P_SERVICE) as WifiP2pManager?
```

Comment:
so irritating that the emulator is just being stupid. Perhaps this is a rare time to add a comment just in case somebody comes along and says "What surely that service can't be null"

Reply:
Actually, this should've been a lazy init right from the beginning; but why we used a `lateinit`?

Reply:
I believe this was a several hundred comment review with the GSOC participant and I didn't want to delay further by pressing the point that this should actually come from Dagger and then during the conversion to kotlin I probably also judged it inappropriate or just plain overlooked it